python/lsst/daf/butler/core/config.py

```diff
@@ -353,7 +456,13 @@ def _processExplicitIncludes(self):
         # Search paths for config files
         searchPaths = [os.path.curdir]
         if self.configFile is not None:
-            searchPaths.append(os.path.abspath(os.path.dirname(self.configFile)))
+            if isinstance(self.configFile, str):
```

Comment:
If you could turn `str` into `ButlerURI` earlier you could get rid of this case and the one below.  But if you can turn both into a `Resource` that would be even better.

Reply:
I can't turn an S3 URI into a `Resource` but I should be able to convert all the plain strings into schemeless URIs.  I had thought about adding a pkg_resources URI scheme to ButlerURI but ButlerURI can't do concrete things yet (like check for existence).

Reply:
I think this is the same as Jim's ABC comment above.  We're not talking about `Resource` as currently defined but a new `Resource` ABC.